src/Functions/seriesOutliersDetectTukey.cpp

```diff
@@ -0,0 +1,283 @@
+                K = k_val;
+            else
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "The fifth argument of function {} must be a positive number", getName());
+        }
+
+        ColumnPtr col_res;
+
+        if (executeNumber<UInt8>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<UInt16>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<UInt32>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<UInt64>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<Int8>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<Int16>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<Int32>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<Int64>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<Float32>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res)
+            || executeNumber<Float64>(arr_data, arr_offsets, min_percentile, max_percentile, K, col_res))
+        {
+            return col_res;
+        }
+        else
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "Illegal column {} of first argument of function {}",
+                arguments[0].column->getName(),
+                getName());
+    }
+
+private:
+    template <typename T>
+    bool executeNumber(
+        const IColumn & arr_data,
+        const ColumnArray::Offsets & arr_offsets,
+        Float64 min_percentile,
+        Float64 max_percentile,
+        Float64 K,
+        ColumnPtr & res_ptr) const
+    {
+        const ColumnVector<T> * src_data_concrete = checkAndGetColumn<ColumnVector<T>>(&arr_data);
+        if (!src_data_concrete)
+            return false;
+
+        const PaddedPODArray<T> & src_vec = src_data_concrete->getData();
+
+        auto outliers = ColumnFloat64::create();
+        auto & outlier_data = outliers->getData();
+
+        ColumnArray::ColumnOffsets::MutablePtr res_offsets = ColumnArray::ColumnOffsets::create();
+        auto & res_offsets_data = res_offsets->getData();
+
+        std::vector<Float64> src_sorted;
+
+        ColumnArray::Offset prev_src_offset = 0;
+        for (auto src_offset : arr_offsets)
+        {
+            chassert(prev_src_offset <= src_offset);
+            size_t len = src_offset - prev_src_offset;
+            if (len < 4)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "At least four data points are needed for function {}", getName());
+
+            src_sorted.assign(src_vec.begin() + prev_src_offset, src_vec.begin() + src_offset);
+            std::sort(src_sorted.begin(), src_sorted.end());
+
+            Float64 q1, q2;
+
+            auto p1 = len * min_percentile;
```

Comment:
Could we please write out the actual data type on the left side, I believe it is `Float64`